src-packed/openaiClientFactory.js

```diff
@@ -0,0 +1,67 @@
+const { Configuration, OpenAIApi } = require("openai");
+
+const configObjectLocalStorageName = "icr-openai-config";
+
+export function setAzureManagedConfig(endpoint, key) {
+  let config = {
+    azureEndpoint: endpoint,
+    azureKey: key
+  };
+
+  window.localStorage.setItem(configObjectLocalStorageName, JSON.stringify(config));
+}
+
+export function setPublicOpenaiConfig(key) {
+  let config = {
+    openaiApiKey: key
+  };
+
+  window.localStorage.setItem(configObjectLocalStorageName, JSON.stringify(config));
+}
+
+export function clearOpenaiConfig() {
+  window.localStorage.removeItem(configObjectLocalStorageName);
+}
+
+export function getOpenaiClient() {
+  let serilizedConfig = window.localStorage.getItem(configObjectLocalStorageName);
+
+  if (!serilizedConfig) {
+    return null;
+  }
+
+  let parsedConfig = JSON.parse(serilizedConfig);
+
+  if (Object.hasOwn(parsedConfig, 'azureEndpoint') && Object.hasOwn(parsedConfig, 'azureKey')) {
+    if (!parsedConfig.azureEndpoint.endsWith("/")) {
+      parsedConfig.azureEndpoint = parsedConfig.azureEndpoint + "/";
+    }
+
+    var epBasePath = `${parsedConfig.azureEndpoint}openai/deployments/icrgpt-35-turbo`
+
+    var openai = new OpenAIApi(new Configuration({
+      apiKey: parsedConfig.azureKey,
+      basePath: epBasePath,
+      baseOptions: {
+        headers: { 'api-key': parsedConfig.azureKey },
+        params: {
+          // https://learn.microsoft.com/azure/cognitive-services/openai/reference#chat-completions
+          'api-version': '2023-03-15-preview'
+        }
+      }
+    }));
```

Comment:
In a future PR, I think we could move this to a new class in `src/packed`?

It would be ok to ship it, the extension doesn't call it yet.

Reply:
I added to the `xtbuild` configuration here:

https://github.com/jonathanpeppers/inclusive-code-reviews-browser/blob/e2e41be7f78d4339e2f81f82ad314efccb70947a/package.json#L53

I could have named it anything, but I couldn't pick `src` as that was the original source code of the spellchecker.